### ISSUE TITLE
Rom Cache System -- Generalized Database.

### ISSUE THREAD

**Issue by richard42** [mupen64plus/mupen64plus-user-issues#75]
Originally reported on Google Code with ID 75

```
From Issue 11:

We need to work out a system where all areas of mupen64plus can have access
to cached rom information, not just the rom browser. That way, the cheat
system can be tied into it also, so it won't have to maintain its own Rom
info (CRC, name, etc).
```

Reported by `ebenblues` on 2008-04-22 21:27:11

**Comment by richard42** [comment]
```
I agree, issues 77, and 47 would also benefit from moving the rom caching 
information from the GUI and putting it in core where all the plugins could access.

Since MD5 is suggested as the mechanism for assigning GoodNames, issue 66 would get

fixed also. 

Just a reference, here are all the cached options from Project64:

Goodname
Country
Size
Comments / Notes (User)
Filename

Catride ID
CRC1
CRC2
CIC Chip
Developer
Filename
Force Feedback
Genre
Internal Name
Notes (Core)
Notes (Plugins)
Players
Release Date

I'm not suggesting to implement all of these. The Developer, and Genre seem 
somewhat silly to me and we don't want to maintain too large a database of items 
not intrinsically in the ROM itself. We also might want to use MD5 instead of dual

CRC to assign GoodNames and hence cheats.
```

Reported by `sknauert@wesleyan.edu` on 2008-04-23 19:42:58

**Comment by richard42** [comment]
```
I wrote up this document to aid us in the specifications of this system. This is
actually a high priority because a lot of things depend on it.
```

Reported by `sgorman07` on 2008-05-08 01:25:21
- **Labels added**: Priority-High
- **Labels removed**: Priority-Medium

<hr>
- _Attachment: [romcachesystem.pdf](https://storage.googleapis.com/google-code-attachments/mupen64plus/issue-75/comment-2/romcachesystem.pdf)_

**Comment by richard42** [comment]
```
After the branch is created you should put a version of this document in ODF or
whatever modifiable format in the doc folder, so that it can be kept up to date.
```

Reported by `richard42g` on 2008-05-08 01:51:43

**Comment by richard42** [comment]
```
I made okaygo's document into a wiki page, rather than keep it in svn:

http://code.google.com/p/mupen64plus/wiki/RomCacheSystem
```

Reported by `ebenblues` on 2008-05-08 20:32:11

**Comment by richard42** [comment]
```
I will be working on this branch... feel free to help out how you see fit!
```

Reported by `sgorman07` on 2008-05-08 22:09:59

**Comment by richard42** [comment]
```
This needs to be done ASAP
```

Reported by `sgorman07` on 2008-05-11 20:19:10
- **Labels added**: Priority-Critical, Milestone-1.4
- **Labels removed**: Priority-High

**Comment by richard42** [comment]
```
Critical priority is for bugs which are breaking builds or causing severe
malfunctions with normal program usage.  This is really a new feature which will fix
a couple of problems.  Normally it would be Medium but since you indicated that you
want to finish it before 1.4, it should be High.
```

Reported by `richard42g` on 2008-05-11 20:37:12
- **Labels added**: Priority-High
- **Labels removed**: Priority-Critical

**Comment by richard42** [comment]
```
I'd say with commit 554 this is Fixed. :) Richard wants to get 1.4 out sooner 
rather than later, so the merge to trunk will occur right after the release. I 
personally disagree, but don't think its important enough to argue. Setting to 
fixed and removing 1.4 milestone.
```

Reported by `sknauert@wesleyan.edu` on 2008-06-13 05:11:09
- **Status changed**: `Fixed`
- **Labels removed**: Milestone-1.4
